Source/FullScreenPokemon.ts

```diff
@@ -6667,6 +6667,95 @@ module FullScreenPokemon {
+            let dt: number = 1;
+            let direction: number = defenderName === "opponent" ? -1 : 1;
+            let scratchLine: string = "ScratchLine";
+            let startX: number;
+            let startY: number;
+            let differenceX: number = defender.width / 2 * FSP.unitsize;
+            let lineArray: IThing[] = [];
+            let menu: IMenu = <IMenu>FSP.MenuGrapher.getMenu("BattleDisplayInitial");
+
+            if (direction === -1) {
+                startX = menu.right - defender.width / 2 * FSP.unitsize;
+                startY = menu.top;
+            } else {
+                startX = menu.left + defender.width * FSP.unitsize;
+                startY = menu.bottom - (defender.height + 8) * FSP.unitsize;
+            }
+
+            let scratch1: IThing = FSP.addThing("ExplosionSmall", startX, startY);
+            let offset: number = scratch1.width * FSP.unitsize / 2;
+            let scratch2: IThing = FSP.addThing("ExplosionSmall", startX + offset * direction * -1, startY + offset);
+            let scratch3: IThing = FSP.addThing("ExplosionSmall", startX + offset * direction * -2, startY + offset * 2);
+
+            FSP.TimeHandler.addEventInterval(
+                function (): void {
+                    let line: IThing;
+                    let left: number = direction === -1 ? scratch1.left : scratch1.right - 3 * FSP.unitsize;
+                    let top: number =  scratch1.bottom - 3 * FSP.unitsize;
+
+                    setTimeout(FSP.shiftHoriz, dt, scratch1, differenceX * direction / 16);
+                    setTimeout(FSP.shiftVert, dt, scratch1, differenceX / 16);
+                    line = FSP.addThing(scratchLine, left, top);
+                    if (direction === 1) {
+                        FSP.flipHoriz(line);
+                    }
+                    lineArray.push(line);
+
+                    left = direction === -1 ? scratch2.left : scratch2.right - 3 * FSP.unitsize;
+                    top =  scratch2.bottom - 3 * FSP.unitsize;
+                    setTimeout(FSP.shiftHoriz, dt, scratch2, differenceX * direction / 16);
+                    setTimeout(FSP.shiftVert, dt, scratch2, differenceX / 16);
+                    line = FSP.addThing(scratchLine, left, top);
+                    if (direction === 1) {
+                        FSP.flipHoriz(line);
+                    }
+                    lineArray.push(line);
+
+                    left = direction === -1 ? scratch3.left : scratch3.right - 3 * FSP.unitsize;
+                    top =  scratch3.bottom - 3 * FSP.unitsize;
+                    setTimeout(FSP.shiftHoriz, dt, scratch3, differenceX * direction / 16);
+                    setTimeout(FSP.shiftVert, dt, scratch3, differenceX / 16);
+                    line = FSP.addThing(scratchLine, left, top);
+                    if (direction === 1) {
+                        FSP.flipHoriz(line);
+                    }
+                    lineArray.push(line);
```

Comment:
As we mentioned offline: looks like the same logic is getting repeated 3x. Probably better to use a `scratches` array or something like that instead of `scratch1`, `scratch2`, `scratch3`.